packages/ui/hooks/index.ts

```diff
@@ -183,16 +185,19 @@ export function useExpiringLocalPreference<Data = any>(
  */
 export function useLocalPreference<Data = any>(
    key: string,
-   defaultData: Data
+   defaultData: Data,
+   validator: (data: any) => Data | null
 ): [Data, (data: Data) => void] {
    const [data, setData] = React.useState(defaultData);
 
    React.useEffect(() => {
       try {
          const serializedData = localStorage.getItem(key);
          if (serializedData != null) {
-            const data = JSON.parse(serializedData);
-            setData(data as Data);
+            const data = validator(JSON.parse(serializedData));
+            if (data !== null) {
+               setData(data as Data);
```

Comment:
do we need this type validation still if we have a validator?

Reply:
You're right, we don't!